sympy/physics/mechanics/_geometry.py

```diff
@@ -387,7 +390,10 @@ def _point_is_on_surface(self, point: Point) -> bool:
         relative_position = point.pos_from(self.point)
         parallel = relative_position.dot(self.axis) * self.axis
         perpendicular = relative_position - parallel
-        return trigsimp(perpendicular.magnitude()) == sqrt(self.radius**2)
+        point_radius = _cancel_sqrt_of_squared_positive(
+            trigsimp(perpendicular.magnitude())
```

Comment:
I agree with Timo's note on this. We do our best to avoid using simplification methods in library code. If you know what the input equation will be (within some reasonable bounds) then you can include it, but in general if a user can pass a complicated expression into this, then we get major slow downs or simplifications failing to occur at all.

Reply:
In the instances where I've used `trigsimp`, provided that the user is on a happy path (i.e. using points that do indeed lie on geometry surfaces) then the expressions we expect to encounter should be very simple. This is because points _should_ be defined relative to the geometry's centre. I think the usages are okay, but I will keep tabs on this as I build more complex biomechanical model examples using this.

Reply:
I've removed all uses of `trigsimp` except two. In the first I'm using it to replace `sin(x)/cos(x)` with `tan(x)`, and in the second I'm using it to replace `sin(x)**2 + cos(x)**2` with `1`. In both cases, particularly the latter, these should be the only trigonometric terms in the expressions so I'd hope we can use these in library code here without penalty.

Reply:
There are specific lower-level functions for doing each of those simplifications in the fu module:
```python
In [15]: from sympy.simplify.fu import TR2i, TR6

In [16]: TR2i(sin(x)/cos(x))
Out[16]: tan(x)

In [17]: TR6(sin(x)**2 + cos(x)**2)
Out[17]: 1
```
https://docs.sympy.org/latest/modules/simplify/fu.html

If that is precisely what is needed then I expect it would be better to use those directly rather than via trigsimp.

Reply:
Unfortunately these aren't compatible with SymEngine:

```python
# Having done `export USE_SYMENGINE=1` before invoking the REPL
>>> from sympy.core.backend import Symbol, cos, sin
>>> from sympy.simplify.fu import TR2i
>>> x = Symbol('x')
>>> TR2i(sin(x)/cos(x))
Traceback (most recent call last):
  ...
AttributeError: 'sin' object has no attribute 'as_powers_dict'

Reply:
This is really limiting. Maybe we should avoid importing from backend.

Reply:
I'm inclined to say that the two instances of `trigsimp` that we have in this PR are safe and shouldn't cause performance problems because we know that the expressions that they're being used on are defined within the constraints of the (simple) geometries we're doing calculations on.

Reply:
We could change that I guess. The definition of `as_powers_dict` is basically:
```python
def as_powers_dict(expr):
    powers = defaultdict(list)
    for arg in Mul.make_args(expr):
        base, exp = arg.as_base_exp()
        powers[base].append(exp)
    return {base: Add(*exponents) for base, exponents in powers.items()}
```
In the mean time could open an issue with SymEngine.

Reply:
I've opened https://github.com/symengine/symengine.py/issues/439 in relation to this.